src/basic/pidref.c

```diff
@@ -71,6 +74,10 @@ bool pidref_equal(PidRef *a, PidRef *b) {
                 if (a->pid != b->pid)
                         return false;
 
+                /* If one is remote and the other isn't, they are not the same */
+                if (pidref_is_remote(a) != pidref_is_remote(b))
```

Comment:
So hmm, below we'd fail to acquire pidfd id with EREMOTE. i think it shall be tweaked to specialize EREMOTE, or add a comment.

As mentioned, we don't really expect any process tracking logic to use this, but just as inspection material, which is relatively OK, but deserve some explanations